### PR TITLE
fix 'source' parameter for request body

### PR DESCRIPTION
**Pull request by mickeyn**
for #522 

not sure if that's a complete solution, but it works for me.

### CODE REVIEW

File: lib/MetaCPAN/Server/Controller.pm [hunk 1]
```diff
@@ -112,7 +112,7 @@ sub search : Path('_search') : ActionClass('Deserialize') {
                 {
                     index => $c->model('CPAN')->index,
                     type  => $self->type,
-                    body  => $c->req->data,
+                    body  => $c->req->data || delete $params->{source},
```
Comment (oalders) [inline lib/MetaCPAN/Server/Controller.pm]: Wow, that seems easy.  :)

**Comment by oalders** [comment]
Is this still a WIP?

**Comment by mickeyn** [comment]
I just wanted it tested to know it solves the original issue reported, but I don't mind having it merged.

**Comment by mickeyn** [comment]
ok, since it does solve the problem of passing the request body through 'source' over GET - we should merge it.
original issue also have ES syntax problems that need to fix fixed on the client end separately.

**Comment by oalders** [comment]
Ok.  Will merge, but I'll add an issue to create a test for this.

**Pull request merged**